### ISSUE TITLE
observeCollection bug

### ISSUE THREAD

**Issue by catx23** [SitePen/dstore#171]
When re-using the same store (with Model) on another grid, the collection.emit('update,{
 target:item
})

is not re-rendering the row anymore! 

The first time I use that store on a grid, this is all working as expected. But when I properly destroy the grid and re-create it, none of the handlers in _StoreMixin._observeCollection are called anymore. I tried all sorts of things but no. 

This is how I forward observe to collection.emit('update') : 

item.property(property).observe(function (value) {
//this is called always correctly

//will trigger only the handlers in observeCollection in the first grid-instance, but not in the second
collection.emit('update,{
  target:item
})

});

Well, no idea. the store is composed out of trackable, cache and memory.

I am using the latest versions of dstore/dgrid/dmodel

**Comment by gtk-aschmidt** [comment]
I'm sure the original poster has long since resolved his problem. But I found this bug report trying to solve my own problem. I've found a solution that works for me. I don't know why. But it might help someone else.

This can be tracked:

declare ([Rest,SimpleQuery,Trackable]))

This can be tracked:

Cache.create(declare ([Rest,SimpleQuery,Trackable])))

This can NOT be tracked:

declare ([Rest,Cache,SimpleQuery,Trackable])

**Comment by denov** [comment]
this works for me if you call .track() on the collection.  but this has the issue you wrote in #188

```
        var tracked = results.track();
        when(tracked.forEach(insertReference, this), function(){ defer.resolve('references in ', this.id); });
```

**Comment by msssk** [comment]
This sounds like a duplicate of SitePen/dgrid#1397 which was fixed in SitePen/dgrid@34e85b8. If you find an issue with dstore please create a new issue with a description of how to reproduce the problem.
